services/pull/merge.go

```diff
@@ -23,14 +23,68 @@ import (
 	"code.gitea.io/gitea/modules/cache"
 	"code.gitea.io/gitea/modules/git"
 	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/markup/markdown"
 	"code.gitea.io/gitea/modules/notification"
 	"code.gitea.io/gitea/modules/references"
 	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
 	"code.gitea.io/gitea/modules/timeutil"
 	asymkey_service "code.gitea.io/gitea/services/asymkey"
 	issue_service "code.gitea.io/gitea/services/issue"
 )
 
+// GetDefaultMergeMessage returns default message used when merging pull request
+func GetDefaultMergeMessage(baseGitRepo *git.Repository, pr *models.PullRequest, mergeStyle repo_model.MergeStyle) string {
+	if err := pr.LoadHeadRepo(); err != nil {
+		log.Error("LoadHeadRepo[%d]: %v", pr.HeadRepoID, err)
+		return ""
+	}
+	if err := pr.LoadIssue(); err != nil {
+		log.Error("Cannot load issue %d for PR id %d: Error: %v", pr.IssueID, pr.ID, err)
+		return ""
+	}
+	if err := pr.LoadBaseRepo(); err != nil {
+		log.Error("LoadBaseRepo: %v", err)
+		return ""
+	}
+
+	issueReference := "#"
+	if pr.BaseRepo.UnitEnabled(unit.TypeExternalTracker) {
+		issueReference = "!"
+	}
+
+	if mergeStyle != "" {
+		templateFilepath := fmt.Sprintf(".gitea/MERGE_MESSAGE_%s_TEMPLATE.md", mergeStyle)
```

Comment:
If I see that correctly, this would lead to files such as `.gitea/MERGE_MESSAGE_rebase-merge_TEMPLATE.md`.
Is that really intended?

I can see a couple alternatives:
1. `.gitea/<strings.toUpper(mergeStyle)>_MESSAGE_TEMPLATE.md`
2. `.gitea/MERGE_MESSAGE_<strings.toUpper(mergeStyle)>_TEMPLATE.md`(the current one but uppercased). I don't prefer this one as there is a redundancy in the name (i.e. `MERGE_MESSAGE_REBASE-MERGE_TEMPLATE.md`)
3. `.gitea/<mergestyle>_message_template.md` (same as 1. but in lowercase)

Reply:
Either way, this feature should be documented (and the PR title should be updated) once we agree on a naming strategy. Otherwise, no one knows these template files can exist and are supported.